extensions/network/configurationinternal.m

```diff
@@ -300,6 +300,97 @@ static int dynamicStoreLocalHostName(lua_State *L) {
     return 1 ;
 }
 
+// internal stuff to make setLocation work
+#define kSCPreferencesOptionChangeNetworkSet    CFSTR("change-network-set") // CFBooleanRef
+SCPreferencesRef
+SCPreferencesCreateWithOptions      (
+                                     CFAllocatorRef      allocator,
+                                     CFStringRef     name,
+                                     CFStringRef     prefsID,
+                                     AuthorizationRef    authorization,
+                                     CFDictionaryRef options
+                                     );
+
+/// hs.network.configuration:setLocation(location) -> boolean
+/// Method
+/// Switches to a new location
+///
+/// Parameters:
+///  * location - string containing UUID or name of new location
+///
+/// Returns:
+///  * bool
+static int dynamicStoreSetLocation(lua_State *L) {
+    LuaSkin *skin = [LuaSkin shared] ;
+    [skin checkArgs:LS_TUSERDATA, USERDATA_TAG, LS_TSTRING, LS_TBREAK];
+    
+    NSString *target ;
+    if (lua_gettop(L) == 2) {
```

Comment:
Unless you also mark the `LS_TSTRING` as optional (i.e. `LS_TSTRING | LS_TOPTIONAL`), checking the stack top with `lua_gettop(L)` isn't necessary -- it will always be 2 if it's made it that far.  So... if `nil` is a valid value for `target`, then add `| LS_TOPTIONAL` to the checkArgs method for the LS_TSTRING argument; if it isn't, then line 328 (and the corresponding close brace) isn't necessary but isn't going to crash either...